src/schema.ts

```diff
@@ -51,7 +54,33 @@ export class Schema {
       };
     });
 
-    return new Schema(fieldSchemas);
+    let schema = new Schema(fieldSchemas);
+
+    // calculate preset bins
+    for (let fieldSchema of fieldSchemas) {
+      if (fieldSchema.type === Type.QUANTITATIVE) {
+        fieldSchema.binStats = {};
+        for (let maxbins of opt.maxBinsList) {
+          fieldSchema.binStats[maxbins] = binSummary(maxbins, fieldSchema.stats);
+        }
+      } else if (fieldSchema.type === Type.TEMPORAL) {
+        // need to get min/max of date data
+        fieldSchema.stats.min = new Date(data[0][fieldSchema.field]);
+        fieldSchema.stats.max = new Date(data[0][fieldSchema.field]);
+        for (var i = 0; i < data.length; i++) {
+          const time = new Date(data[i][fieldSchema.field]).getTime();
+          if (time < (fieldSchema.stats.min as Date).getTime()) {
+            fieldSchema.stats.min = new Date(time);
```

Comment:
I'm not sure which format we should store date in yet, but i'll let go through for now and we can revisit this later.